### ISSUE TITLE
Move towards manylinux2014 wheels

### ISSUE THREAD

**Issue by garrettwrong** [flatironinstitute/cufinufft#93]
Following #92 , it is time to move to newer OS and CUDA basis for the python wheels.

- [x] Confirm manylinux2014 ci image builds (`make` inside the container)
- [x] Confirm `make check` inside the container succeeds on my machine
- [ ] Confirm container build and check on a second machine
- [x] Confirm build wheel script runs
- [x] Check each wheel output 
- [x] Check if python 3.5 is deprecated; if so purge it
- [x] Update the ci/readme to be a little more scriptable (vars for version etc). Perhaps make it a real script and just have CI call the script?.. (probably)
- [x] Consider what the `next thing` after 2014 will be (manylinux, OS, CUDA version) so project is prepared again.

**Comment by garrettwrong** [comment]
Confirmed Python 3.5 is end of life (https://www.python.org/dev/peps/pep-0478/).  It appears `pip` itself will be dropping support for it based on this note I found testing the 2014 3.5 wheel:

```
DEPRECATION: Python 3.5 reached the end of its life on September 13th, 2020. Please upgrade your Python as Python 3.5 is no longer maintained. pip 21.0 will drop support for Python 3.5 in January 2021. pip 21.0 will remove support for this functionality.
```

I'll put together a small change set removing 3.5 from the build script along with any changes needed for 2014 wheel support (had some trouble with 3.7, which I think is why I didn't use these wheels in the summer; memories starting to come back to me).

As far as the `next thing`, it doesn't appear to have been decided in a PEP yet.  2014 is claimed to go through 2024. I can ask around, but I don't know there is much else we can prepare for yet.
